src/System.Runtime.Extensions/src/System/Environment.Windows.cs

```diff
@@ -155,7 +157,10 @@ public static string SystemDirectory
 
                 if (requiredSize > buffer.Length)
                 {
-                    buffer = new char[requiredSize];
+                    if (requiredSize < SystemDirectoryAllocLimit)
+                        buffer = stackalloc char[requiredSize];
+                    else
+                        buffer = new char[requiredSize];
```

Comment:
Why not just increase 32 => 260 above?

Reply:
With what you propose it would always allocate 260 characters. Worst case, the path is 261 chars long, it would first stackalloc 260 chars and afterwards 261 chars on the heap. 

With my change it would always first allocate 32 chars on the stack and if that's not enough the required amount smaller than 260. E.g. the path is 33 chars long, it would first allocate 32 chars and afterwards 33 chars (65 in total).

Reply:
Not sure what is better, 2 small stackallocs (1 fixed size, 1 variable sized) or 1 big stackalloc (fixed size)

Reply:
Do you see a performance difference between stackalloc'ing 32 chars vs 260 chars?  This assembly now has the initlocals flag cleared, so it's not zero'ing out the stack memory.  I wouldn't expect there to be any measurable difference.

Reply:
right, not zero'ing the memory out is perfect here. I will change it to 260.